plugins/inputs/elasticsearch/elasticsearch.go

```diff
@@ -136,12 +161,27 @@ func (e *Elasticsearch) Gather(acc telegraf.Accumulator) error {
 			} else {
 				url = s + statsPath
 			}
+			e.isMaster = false
+
+			if e.ClusterStats {
+				// get cat/master information here so NodeStats can determine
+				// whether this entrance is on the Master
+				e.setCatMaster(s + catMasterPath)
```

Comment:
The existing code had a unused const heathPath. I simply followed the pattern used by that statsPath, to have these calls use their constant values. However, I see your point - the code always uses the const statspath, and instead of using constants for the others (which are optional), it could instead use hardcoded strings when the options are detected. I can make that change if you'd prefer.